### PR TITLE
Implement a finer grained check for the existince of APIs

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Bugfix

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by ctron** [comment]
@enmasse-ci run tests profile=smoke

**Comment by ctron** [comment]
@enmasse-ci run tests profile=shared

**Comment by k-wall** [comment]
Running the unit tests with `make test_go` panics with this change unless the environment variable NAMESPACE is not set.  I don't think the average developer would have this in their environment.

panic: infrastructure namespace is not set in 'NAMESPACE'

```
goroutine 1 [running]:
github.com/enmasseproject/enmasse/pkg/util.init.0()
	/home/keith/go/src/github.com/enmasseproject/enmasse/pkg/util/env.go:101 +0x111

```

**Comment by ctron** [comment]
> @ctron Running the unit tests with `make test_go` panics with this change unless the environment variable NAMESPACE is not set. I don't think the average developer would have this in their environment.
> 
> panic: infrastructure namespace is not set in 'NAMESPACE'
> 
> ```
> goroutine 1 [running]:
> github.com/enmasseproject/enmasse/pkg/util.init.0()
> 	/home/keith/go/src/github.com/enmasseproject/enmasse/pkg/util/env.go:101 +0x111
> ```

Same for the CI *sigh* …

File: pkg/util/openshift.go [hunk 1]
```diff
@@ -1,68 +1,90 @@
...
-	value, ok := os.LookupEnv("ENMASSE_OPENSHIFT")
+func detectApi(gvk schema.GroupVersionKind) bool {
+
+	name := fmt.Sprintf("ENMASSE_HAS_API_%s_%s_%s", strings.ToUpper(gvk.Group), strings.ToUpper(gvk.Version), strings.ToUpper(gvk.Kind))
+
+	value, ok := os.LookupEnv(name)
 	if ok {
-		log.Info("Set by env-var 'ENMASSE_OPENSHIFT': " + value)
+		log.Info("Set by env-var '" + name + "': " + value)
 		return strings.ToLower(value) == "true"
```
Comment (k-wall) [inline pkg/util/openshift.go]: Should we be using `strconv.ParseBool()` (and treating error as false)?  A user familiar to Go conventions could pass one of the other Go truthy values (1, T, 0, F)?

**Comment by ctron** [comment]
@enmasse-ci run tests profile=shared

File: pkg/util/openshift.go (hunk 1 above)
Comment (ctron) [reply pkg/util/openshift.go]: I kept that line as it was, but I agree, that might be a nice change.

Comment (ctron) [reply pkg/util/openshift.go]: Just checked on the API, that would require us to handle `error` was well … which is different than the Java side

**Pull request merged**